pkg/engine/planner/physical/optimizer.go

```diff
@@ -248,19 +277,86 @@ func (r *projectionPushdown) applyProjectionPushdown(node Node, projections []Co
 			}
 		}
 		return changed
+	case *ParseNode:
+		// Only apply the pushdown for Metric queries. Log queries should request all keys
+		if !r.isMetricQuery() || len(ambiguousProjections) == 0 {
+			return false
+		}
```

Comment:
> Log queries should request all keys

This may not be true forever, so I don't think we want to enforce that logic here. The DataObjScan case uses `len(node.Projections) == 0 && applyIfNotEmpty` to represent this logic; can we use the same here? 

(Then we could remove isMetricQuery)